instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java

```diff
@@ -81,17 +81,11 @@ public Context startSpan(
     onConnectionAndRequest(spanBuilder, connection, request);
 
     Context context = withServerSpan(parentContext, spanBuilder.startSpan());
-    context = customizeContext(context, request);
     attachServerContext(context, storage);
 
     return context;
   }
 
-  /** Override in subclass to customize context that is returned by {@code startSpan}. */
-  protected Context customizeContext(Context context, REQUEST request) {
```

Comment:
The idea behind this was that it allows modifying context before it gets attached to the carrier so that `RequestDispatcherAdvice` that reads context from request attribute would get the same naming context as the request originally had.

Reply:
I pulled this code up into `startSpan()` so that `ServletHttpServerTracer` can call `ServerSpanNaming.init()` with `SERVLET` or `CONTAINER` based on whether it is called form a servlet or filter.

The tests haven't all passed yet so not sure if it was a success or no 😄.

Reply:
oh, ya, we need to add `ServletSpanNaming` to the context before attaching the context to the `STORAGE` param 👍

Reply:
brought back `customizeContext()` and added a `@Deprecated` method in `ServerSpanNaming` to allow `ServletHttpServerTracer` to update the `source` separately from calling `init` and we can revisit to see if we can do both at once after we migrate to the new Instrumenter API